### ISSUE TITLE
Horizontal drag inside TextField breaks all single tap gestures

### ISSUE THREAD

**Issue by Adam-Langley** [flutter/flutter#153735]
### Steps to reproduce

1. Wrap a TextField inside a ListView
2. Run app on iOS
3. type some text into text field
4. drag horizontally across the text field, as if you are "striking out" a word, then release. If the "caret position magnifier" appears, then you held down for too long before starting the drag movement - this will not trigger the bug. You must begin the drag more quickly.
5. now tap again, anywhere in the text field, as if to move the caret position

### Expected results

Caret position should be placed at the tap location, or you get the "select all" context menu - depends on where you tapped

### Actual results

Nothing happens.
Exception stack trace is generated.
Text Field single-tap gestures are now totally broken until an aggressive rebuild happens.

```
flutter: [E] TIME: 2024-08-20T13:19:11.522688 Unhandled Exception: 'package:flutter/src/gestures/tap_and_drag.dart': Failed assertion: line 1017 pos 14: '_dragState == _DragState.possible': is not true.  ERROR: 'package:flutter/src/gestures/tap_and_drag.dart': Failed assertion: line 1017 pos 14: '_dragState == _DragState.possible': is not true.  STACKTRACE: #0      _AssertionError._doThrowNew (dart:core-patch/errors_patch.dart:50:61)
#1      _AssertionError._throwNew (dart:core-patch/errors_patch.dart:40:5)
errors_patch.dart:40
#2      BaseTapAndDragGestureRecognizer.acceptGesture (package:flutter/src/gestures/tap_and_drag.dart:1017:14)
tap_and_drag.dart:1017
#3      GestureArenaManager.sweep (package:flutter/src/gestures/arena.dart:169:27)
arena.dart:169
#4      GestureBinding.handleEvent (package:flutter/src/gestures/binding.dart:505:20)
binding.dart:505
#5      GestureBinding.dispatchEvent (package:flutter/src/gestures/binding.dart:481:22)
binding.dart:481
#6      RendererBinding.dispatchEvent (package:flutter/src/rendering/binding.dart:450:11)
```

### Code sample

sample code
```
ListView(children: [TextField()]),
```

I've tracked the "bad internal state" down to the `BaseTapAndDragGestureRecognizer`.
The field `_primaryPointer` should be null when the tap occurs, however, the previous exception has left the previous _primaryPointer hanging around, hence `isPrimaryPointerAllowed` on line 940, returns false, when it should return true.

```
  @override
  bool isPointerAllowed(PointerEvent event) {
    if (_primaryPointer == null) {
```

### Screenshots or Video

https://github.com/user-attachments/assets/4b02dd9f-a372-4d2b-bc9c-f5469cfc2740

### Logs

```
flutter: [E] TIME: 2024-08-20T13:19:11.522688 Unhandled Exception: 'package:flutter/src/gestures/tap_and_drag.dart': Failed assertion: line 1017 pos 14: '_dragState == _DragState.possible': is not true.  ERROR: 'package:flutter/src/gestures/tap_and_drag.dart': Failed assertion: line 1017 pos 14: '_dragState == _DragState.possible': is not true.  STACKTRACE: #0      _AssertionError._doThrowNew (dart:core-patch/errors_patch.dart:50:61)
#1      _AssertionError._throwNew (dart:core-patch/errors_patch.dart:40:5)
errors_patch.dart:40
#2      BaseTapAndDragGestureRecognizer.acceptGesture (package:flutter/src/gestures/tap_and_drag.dart:1017:14)
tap_and_drag.dart:1017
#3      GestureArenaManager.sweep (package:flutter/src/gestures/arena.dart:169:27)
arena.dart:169
#4      GestureBinding.handleEvent (package:flutter/src/gestures/binding.dart:505:20)
binding.dart:505
#5      GestureBinding.dispatchEvent (package:flutter/src/gestures/binding.dart:481:22)
binding.dart:481
#6      RendererBinding.dispatchEvent (package:flutter/src/rendering/binding.dart:450:11)
```

### Flutter Doctor output

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.24.0, on macOS 14.0 23A344 darwin-x64, locale en-NZ)
[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.3)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2023.2)
[✓] VS Code (version 1.92.2)
[✓] Connected device (4 available)
[✓] Network resources

• No issues found!
```

**Comment by huycozy** [comment]
Hi @Adam-Langley 
I think this can be reproduced more easily with steps at https://github.com/flutter/flutter/issues/153458. The crash stack trace is similar as well. It's confirmed as fixed on master channel so please switch to master channel and retry.

I will mark this as a duplicate of https://github.com/flutter/flutter/issues/153458. If you still experience this on master channel with the given steps, please write in the comment and I will reopen this. Thanks!

**Comment by Adam-Langley** [comment]
@huycozy - confirmed, fixed on master. Thank you.
